### DISCUSSION TITLE
Using / installing the TextToWaveform models

### DISCUSSION THREAD

**Discussion by dln22** [NVIDIA-NeMo/NeMo#5113] [Q&A] [answered]
Hello,

Could someone please advise on how to install the `TextToWaveform` models. It seems when one do:

`python -m pip install nemo_toolkit[all]`

the `TextToWaveform` module is not added. According to the docs (https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/tts/api.html#nemo.collections.tts.models.base.TextToWaveform) it should be possible to e.g. import:

```python
from nemo.collections.tts.models.base import TextToWaveform
```

I am asking as I would love to use the `FastPitch_HifiGan_E2E` model.

Best wishes,
Daniel

**Comment by redoctopus** [comment, marked as answer]
Hi, the `FastPitch_HifiGan_E2E` model was deprecated in release 1.9.0, so you'll either want to use version 1.8.0 or try using a `FastPitchModel` and passing the output to a `HifiGanModel`:

```
# Load FastPitch
from nemo.collections.tts.models import FastPitchModel
spec_generator = FastPitchModel.from_pretrained("tts_en_fastpitch") # or .restore_from("<ckpt_path>")

# Load HiFi-GAN
from nemo.collections.tts.models import HifiGanModel
vocoder_model = HifiGanModel.from_pretrained(model_name="tts_hifigan") # or .restore_from("<ckpt_path>")

# Generate audio
import soundfile as sf
parsed = spec_generator.parse("Some test sentence here.")
spectrogram = spec_generator.generate_spectrogram(tokens=parsed)
audio = vocoder_model.convert_spectrogram_to_audio(spec=spectrogram)
```

Apologies--our documentation is somewhat out of date, but we're actively working on updating it!

**Reply by dln22** [reply]
Super thanks for the answer! :)
